corehq/apps/api/resources/v0_5.py

```diff
@@ -1002,7 +1004,10 @@ def dispatch(self, request_type, request, **kwargs):
         if not toggles.ODATA.enabled_for_request(request):
             raise ImmediateHttpResponse(response=HttpResponseNotFound('Feature flag not enabled.'))
         self.config_id = kwargs['config_id']
-        return super(ODataCaseResource, self).dispatch(request_type, request, **kwargs)
+        with TimingContext() as timer:
+            response = super(ODataCaseResource, self).dispatch(request_type, request, **kwargs)
+        record_feed_access_in_datadog(request, self.config_id, timer.duration, response)
```

Comment:
I don't know much about context managers, but it seems weird that you can access `timer` outside of the `with` block. Is that common?

Reply:
Yes: https://stackoverflow.com/questions/6432355/variable-defined-with-with-statement-available-outside-of-with-block

Here's an example from HQ: https://github.com/dimagi/commcare-hq/blob/fa4115037248979d925dc067994fe5234127125e/corehq/apps/couch_sql_migration/couchsqlmigration.py#L129

Reply:
interesting, thanks for the lesson 🙂